classes/capquiz.php

```diff
@@ -82,6 +82,10 @@ public function is_grading_completed() : bool {
         return $this->record->timedue < time() && $this->record->timedue > 0;
     }
 
+    public function duedate() : string {
+        return userdate($this->record->timedue, get_string('strftimedatetime', 'langconfig'));
+    }
+
```

Comment:
The `capquiz` class is only meant to keep data about a CAPQuiz (mostly data mappings). I think this function should be removed, and the logic moved to the `question_attempt_renderer.php` file.
```suggestion
```